### PR TITLE
feat: Add values to env then use them.

### PR DESCRIPTION
**Pull request by clemens-tolboom**
## Related

- [x] ? Does it block `release-test.R`ing https://github.com/molgenis/molgenis-service-armadillo/pull/277
  - See comment https://github.com/molgenis/molgenis-service-armadillo/pull/486#pullrequestreview-1658978284 

## Summary

Trying to run an R script the first time all those questions are time consuming.

Let us use `.env` mechanism available.

Is it a Chore or a Feat?

### CODE REVIEW

File: scripts/release/dev.env.dist
Comment (clemens-tolboom) [inline scripts/release/dev.env.dist:1]: Needs self documentation.

To enable copy to dev.env

**Comment by mswertz** [comment]
converted to draft because work in progress? also I think our version release plugin requires the 'feat' to be lowercase in title.

File: scripts/release/dev.env.dist [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Value of 'x' will trigger downloading test file
+# This is actually the parent of the checkout dir without /molgenis-service-armadillo
+GIT_CHECKOUT_DIR=x
```
Comment (SietsmaRJ) [inline scripts/release/dev.env.dist:3]: I'd say `GIT_CLONE_PATH`

File: scripts/release/dev.env.dist [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+RDA_TEST_DIR=
+
+PROFILE=xenon
+ARMADILLO_URL=http://localhost:8080
+ADMIN_PASSWORD=admin
+
+OIDC_EMAIL=your-email-is-invalid
+
+# The token will invalidate probably in 1 hour so replace it or set to empty
+TOKEN=
```
Comment (SietsmaRJ) [inline scripts/release/dev.env.dist:17]: For the entirety of the .env.dist file:

Set all variables to the `release-test.R` default, so for instance: 
- `GIT_CLONE_PATH=~/git`
- `RDA_FILE_PATH=~/git/molgenis-service-armadillo/data/testing`
- PROFILE, ARMADILLO_URL and ADMIN_PASSWORD are already set properly
- `OIDC_EMAIL=` (this has to be set by the user before running the tests)

With this, we no longer have to ask the user to enter anything anymore, just present the user with the variable set in the .env.dist config file, with a simple `enter to confirm` (if escape or so, stop running the code)

File: scripts/release/release-test.R [hunk 3]
```diff
@@ -461,17 +469,36 @@ if(url.exists(armadillo_url)) {
...
 
-cat("Location of molgenis-service-armadillo on your PC (for default ~/git/, press enter, if not available press x): ")
-service_location <- readLines("stdin", n=1)
+service_location = Sys.getenv("GIT_CHECKOUT_DIR")
+if(service_location == ""){
+  cat("Location of molgenis-service-armadillo on your PC (for default ~/git/, press enter, if not available press x): ")
+  service_location <- readLines("stdin", n=1)
+} else {
+   cat("GIT_CHECKOUT_DIR from '.env' file\n")
+}
```
Comment (SietsmaRJ) [inline scripts/release/release-test.R:478]: See comment on entirety of dev.env.dist

File: scripts/release/release-test.R [hunk 4]
```diff
@@ -461,17 +469,36 @@ if(url.exists(armadillo_url)) {
...
 
-cat("Enter password for admin user (basic auth)\n")
-admin_pwd<- getPass::getPass()
+admin_pwd = Sys.getenv("ADMIN_PASSWORD")
+if(service_location == ""){
+  cat("Enter password for admin user (basic auth)\n")
+  admin_pwd<- getPass::getPass()
+} else {
+  cat("ADMIN_PASSWORD from '.env' file\n")
+}
```
Comment (SietsmaRJ) [inline scripts/release/release-test.R:488]: See comment on entirety of dev.env.dist

File: scripts/release/release-test.R [hunk 5]
```diff
@@ -461,17 +469,36 @@ if(url.exists(armadillo_url)) {
...
 
-cat("Enter your OIDC username (email): ")
-user <- readLines("stdin", n=1)
+user = Sys.getenv("OIDC_EMAIL")
+if(user == ""){
+  cat("Enter your OIDC username (email): ")
+  user <- readLines("stdin", n=1)
+} else {
+   cat("OIDC_EMAIL from '.env' file\n")
+}
```
Comment (SietsmaRJ) [inline scripts/release/release-test.R:496]: See comment on entirety of dev.env.dist

File: scripts/release/release-test.R [hunk 6]
```diff
@@ -461,17 +469,36 @@ if(url.exists(armadillo_url)) {
...
 
 if(service_location == "") {
+  # FIXME: as we are in same git repo path could be relative to this path?
+  # ../../data/shared-lifecycle/
+  # guess we have to download/copy to a tmp directory before uploading to armadillo.
```
Comment (SietsmaRJ) [inline scripts/release/release-test.R:501]: Completely agree, use the `.parquet` test files by default stored in `../../data/shared-lifecycle`

File: scripts/release/release-test.R [hunk 7]
```diff
@@ -513,22 +540,30 @@ if(service_location == "") {
...
-  download_dir <- add_slash_if_not_added(download_dir)
-  if(dir.exists(download_dir)){
-    cli_alert_info(sprintf("Downloading %s into %s", test_resource, download_dir))
-    download.file("https://github.com/isglobal-brge/brge_data_large/raw/master/data/gse66351_1.rda", paste0(download_dir, test_resource))
-    rda_dir = paste0(download_dir, test_resource)
+rda_dir = user = Sys.getenv("RDA_TEST_DIR")
+if(rda_dir == ""){
+  # FIXME: Similar to above
+  # - we can define a directory for downloading files
+  # - if file 'gse66351_1.rda' does not exists there download it
```
Comment (SietsmaRJ) [inline scripts/release/release-test.R:547]: As all the `.parquet` files should already be available in `<root>/data/shared-lifecycle`, only the RDA file should be downloaded somewhere, which should be defined within the dev.env.dist config file. Maybe check if within the config file the filename is already present, and if not try to check if the path given + filename exists. If not, download.

File: scripts/release/release-test.R [hunk 8]
```diff
@@ -513,22 +540,30 @@ if(service_location == "") {
...
+    rda_dir = readLines("stdin", n=1)
+  } else {
+    cat("Where do you want to download it: ")
+    download_dir = readLines("stdin", n=1)
+    download_dir <- add_slash_if_not_added(download_dir)
+    if(dir.exists(download_dir)){
+      cli_alert_info(sprintf("Downloading %s into %s", test_resource, download_dir))
+      download.file("https://github.com/isglobal-brge/brge_data_large/raw/master/data/gse66351_1.rda", paste0(download_dir, test_resource))
+      rda_dir = paste0(download_dir, test_resource)
+    }
```
Comment (SietsmaRJ) [inline scripts/release/release-test.R:563]: See comment on entirety of dev.env.dist

File: scripts/release/release-test.R [hunk 9]
```diff
@@ -544,8 +579,14 @@ cat("\nAvailable profiles: \n")
...
 
-cat("Which profile do you want to test on? (press enter to continue using xenon; xenon will be created if unavailable) ")
-profile <- readLines("stdin", n=1)
+profile = Sys.getenv("PROFILE")
+if(profile == ""){
+  cat("Which profile do you want to test on? (press enter to continue using xenon; xenon will be created if unavailable) ")
+  profile <- readLines("stdin", n=1)
+} else {
+  cat("PROFILE from '.env' file\n")
+}
```
Comment (SietsmaRJ) [inline scripts/release/release-test.R:588]: See comment on entirety of dev.env.dist

**Review by SietsmaRJ — DISMISSED** [review]

File: scripts/release/dev.env.dist (hunk 2 above)
Comment (clemens-tolboom) [reply scripts/release/dev.env.dist:17]: I was on the CI testing flow.

Settings the defaults sound ok to me but I don't get `~git` and some. We'll've to discuss. I tried in the code using FIXME.

We could add `ASK_FOR_CONFIRMATION=N` but no sure as you have to copy the `dev.env.dist` to `dev.env` or use environment variables to tune this.

I think reading from `dev.env` should actually be `.env` following the dot-dev way. But it is hidden then :-O but CI don't care I guess

File: scripts/release/dev.env.dist [hunk 10]
```diff
@@ -0,0 +1,17 @@
+# Value of 'x' will trigger downloading test file
+# This is actually the parent of the checkout dir without /molgenis-service-armadillo
+GIT_CHECKOUT_DIR=x
+DOWNLOAD_FILE=n
+DOWNLOAD_FILE_NAME=
```
[the remark this answers is not in the document]
Comment (clemens-tolboom) [reply scripts/release/dev.env.dist:5]: Not sure who run `release-test.R`? We have to discuss :)

File: scripts/release/dev.env.dist (hunk 2 above)
Comment (clemens-tolboom) [reply scripts/release/dev.env.dist]: We discussed:
- `*.env files: users have to copy one of the `*.env.dist` files to `.env`. Currently we have one but expect another like `tst.env.dist` or `stage.env.dist` and `ci.env.dist` ... what are [dot env files](https://hexdocs.pm/dotenvy/dotenv-file-format.html)
- as our users have a git checkout most of the time we can stream-line path and files removing Y/N questions

File: scripts/release/release-test.R (hunk 8 above)
Comment (clemens-tolboom) [reply scripts/release/release-test.R:577]: This should be refactored as we will download it when not found in the **//data/** directory or another local/relative directory.

File: scripts/release/release-test.R (hunk 9 above)
Comment (clemens-tolboom) [reply scripts/release/release-test.R:615]: (all seen)

**Comment by clemens-tolboom** [comment]
As this blocks #277 we could accept current status and later refactor.

**Review by clemens-tolboom — COMMENTED** [review]
Not sure the code should have moved down. On master only change was in the `get_from_api`

```
profiles <- get_from_api_with_header("profiles", api_token, auth_type)
```

**Comment by marikaris** [comment]
I think it's a feat :)

**Review by clemens-tolboom — COMMENTED** [review]
Thanks for the addition and changes. LGTM.

File: scripts/release/release-test.R [hunk 11]
```diff
@@ -35,7 +36,12 @@ library(stringi)
 library(MolgenisArmadillo)
 library(DSI)
 library(dsBaseClient)
+
+# FIXME: waiting for PR https://github.com/molgenis/molgenis-r-datashield/pull/62
+#        needed for https://github.com/molgenis/molgenis-service-armadillo/pull/277
+#devtools::install("/Users/clemens/Documents/GitHub/molgenis-r-datashield/")
```
Comment (SietsmaRJ) [inline scripts/release/release-test.R:42]: Is this still required?

Comment (clemens-tolboom) [reply scripts/release/release-test.R:42]: Latest https://github.com/molgenis/molgenis-r-datashield/tags has https://github.com/molgenis/molgenis-r-datashield/pull/62 in it. Not sure how long it takes for CRAN to publish.

https://github.com/molgenis/molgenis-service-armadillo/pull/277 is fixed and we are not yet releasing.

We can let it in and cleanup for the next step. OK?

**Pull request merged**